src/DynamoPackagesUI/Utilities/PackageManagerCommands.cs

```diff
@@ -0,0 +1,503 @@
+        /// Get the folder path to install the package
+        /// </summary>
+        /// <returns></returns>
+        /// TODO: Change the method name
+        public string GetCustomPathForInstall()
+        {
+            string downloadPath = string.Empty;
+
+            downloadPath = RequestGetDownloadPath();
+
+            if (String.IsNullOrEmpty(downloadPath))
+                return string.Empty;
+
+            PackageInstallPath = downloadPath;
+
+            return downloadPath;
+        }
+
+        private string RequestGetDownloadPath()
+        {
+            var args = new PackagePathEventArgs();
+
+            PromptFileSelectionDialog(args);
+
+            if (args.Cancel)
+                return string.Empty;
+
+            return args.Path;
+        }
+
+        private void PromptFileSelectionDialog(PackagePathEventArgs e)
+        {
+            string initialPath = Model.PathManager.DefaultPackagesDirectory;
+
+            e.Cancel = true;
+
+            var errorCannotCreateFolder = PathHelper.CreateFolderIfNotExist(initialPath);
+            if (errorCannotCreateFolder == null)
+            {
+                var dialog = new DynamoFolderBrowserDialog
+                {
+                    // Navigate to initial folder.
+                    SelectedPath = initialPath,
+                    Owner = this.ParentWindow
+                };
+                Application.Current.Dispatcher.Invoke((Action)(() =>
+                {
+                    if (dialog.ShowDialog() == System.Windows.Forms.DialogResult.OK)
+                    {
+                        e.Cancel = false;
+                        e.Path = dialog.SelectedPath;
+                    }
+                }));
+            }
+            else
+            {
+                string errorMessage = string.Format(Wpf.Properties.Resources.PackageFolderNotAccessible, initialPath);
+                System.Windows.Forms.MessageBox.Show(errorMessage, Wpf.Properties.Resources.UnableToAccessPackageDirectory, System.Windows.Forms.MessageBoxButtons.OK, System.Windows.Forms.MessageBoxIcon.Error);
+            }
+        }
+
+        /// <summary>
+        /// Install button click event
+        /// </summary>
+        /// <param name="asset"></param>
+        /// <param name="version"></param>
+        /// <returns>If "cencel" returned to webclient halt the installation. If not cancel then this method returns the | delimitted list of package and its version to be installed</returns>
+        /// TODO: Rename this method to OnInstallPackage
+        public string PackageOnExecuted(dynamic asset, dynamic version)
+        {
+            string downloadPath = this.PackageInstallPath;
+
+            List<Tuple<dynamic, dynamic>> packageVersionData = new List<Tuple<dynamic, dynamic>>();
+
+            string msg = String.IsNullOrEmpty(downloadPath) ?
+                String.Format(Resources.MessageConfirmToInstallPackage, asset["asset_name"], version["version"]) :
```

Comment:
`asset_name`, `dependencies`, `version` etc. are being used as hardcoded strings at many places, you might want to move these to some constants or a separate static class such as `ParameterNames` .